include/simdjson/error.h

```diff
@@ -35,6 +35,7 @@ enum error_code {
   INVALID_URI_FRAGMENT,     ///< Invalid URI fragment
   UNEXPECTED_ERROR,         ///< indicative of a bug in simdjson
   PARSER_IN_USE,            ///< parser is already in use.
+  OUT_OF_ORDER_ITERATION,   ///< tried to iterate an array or object out of order
```

Comment:
optional: tag with on demand so that it is clearer what the error is for.

Reply:
Consider updating https://github.com/simdjson/simdjson/blob/a9480a768bd80174928fee25ecc33947faad4603/src/internal/error_tables.cpp if needed

Reply:
Forget last comment.